### DISCUSSION TITLE
Switching between User Defined / Metered Usage modes (Solix 2 Pro)

### DISCUSSION THREAD

**Discussion by koehntopp** [thomluther/ha-anker-solix#148] [General]
I need to ask a stupid question - I read about all the fancy schedules, but as I have a Shelly 3EM installed, I really only need to switch between charge_priority and detection modes.

I tried in the developer tools, but the mode did not change (Yes, I used the correct user). 
Is that even the right place to do that? How would I switch modes in an automation?

Would be awesome if someone could point me to what I might be doing wrong...

![image](https://github.com/user-attachments/assets/0dd324dd-4aa4-4142-bbb1-24bb84910db2)

**Comment by thomluther** [comment]
Hi @koehntopp, you are asking for something that is not supported by the device.
First of all, that status you listed is just provided as sensor (information only). This device status is furthermore only an integer code from the cloud api (you can see it in the entity attribute).

The api library translates this code into a meaningful description and also introduces extra descriptions for the same code when various conditions are met. For example, the solarbank does not distinguish between charge bypass or charge priority, both have the same status code.
Furthermore those entities are sensors and sensors are not user changeable entitiies types by design, they only provide information.
So anything that you overwrite manually in a sensor is not reflected back to any device, it simply cannot.

Only user changable entity types can be used to reflect user input back to a device and modify something in the device. The integration will use the Api to reflect setting changes back to the solarbank, but the api (and device) must also support this setting change.

And for known settings that are really user changeable via Api, there will be a changeable entitiy implemented in the integration once the setting is undestood and supported by the underlying Api library.

So actually it is not clear what you want to achieve. The solarbank status is managed only by the device firmware and users can only indirectly influence it via modifications of the usage mode or output presets in the App. Those settings are also supported via Api and the integration.

**Comment by koehntopp** [comment]
I knew I was on the wrong path ;)

What I want to do is switch between "only charge" ("Benutzerdefinierter Modus", set to "never discharge") and "discharge according to the data of the smart meter" ("Eigenverbrauch").

Is that possible...? I read the INFO.md a lot of times, there's a screenshot with a "Usage mode" dropdown, but I couldn't figure out the entities to use. Any help appreciated.

**Comment by thomluther** [comment]
If you want to switch only between the 2 modes, that should be supported by the usage mode entity. This however becomes only available when you use the owner account in the integration.
Furthermore the smart meter mode is removed from the selectable options if no smart meter was found in the system response.
It could be that the shelly does not reflect values like the Anker smart meter through the api response and therefore is not recognized as available smart meter. 
As you can see on [supported device list](https://github.com/thomluther/ha-anker-solix?tab=readme-ov-file#supported-sensors-and-devices), the Shelly is not listed.

You would have to [open a feature request ](https://github.com/thomluther/ha-anker-solix/issues) and to use the [export system service](https://github.com/thomluther/ha-anker-solix/blob/main/INFO.md#export-systems-service) to provide a snapshot of your Api responses to verify the api responses with shelly device. Then eventually support for shelly can be added when the required info is available in the known request responses.

**Comment by koehntopp** [comment]
(my bad - got it)

**Comment by koehntopp** [comment]
api_devices.json has the Shelly. What exactly do you need, and where do I send it?

**Reply by thomluther** [reply]
Please [open a feature request ](https://github.com/thomluther/ha-anker-solix/issues/new/choose) and attach your export zip file there. So I can have a look to all responses.

**Comment by thomluther** [comment]
The Shelly as well as User mode switch for Smart meter and Smart Plugs is fully supported with 2.2.0 when logged in as system owner account.
